aws/resource_aws_api_gateway_documentation_part.go

```diff
@@ -0,0 +1,202 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayDocumentationPart() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayDocumentationPartCreate,
+		Read:   resourceAwsApiGatewayDocumentationPartRead,
+		Update: resourceAwsApiGatewayDocumentationPartUpdate,
+		Delete: resourceAwsApiGatewayDocumentationPartDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"location": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"method": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"path": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"status_code": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"properties": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"rest_api_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayDocumentationPartCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	out, err := conn.CreateDocumentationPart(&apigateway.CreateDocumentationPartInput{
+		Location:   expandApiGatewayDocumentationPartLocation(d.Get("location").([]interface{})),
+		Properties: aws.String(d.Get("properties").(string)),
+		RestApiId:  aws.String(d.Get("rest_api_id").(string)),
+	})
+	if err != nil {
+		return err
+	}
+	d.SetId(*out.Id)
+
+	return nil
+}
+
+func resourceAwsApiGatewayDocumentationPartRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	log.Printf("[INFO] Reading API Gateway Documentation Part %s", d.Id())
+	docPart, err := conn.GetDocumentationPart(&apigateway.GetDocumentationPartInput{
+		DocumentationPartId: aws.String(d.Id()),
+		RestApiId:           aws.String(d.Get("rest_api_id").(string)),
+	})
+	if err != nil {
+		if isAWSErr(err, "NotFoundException", "") {
```

Comment:
Nitpick: Constant is available for this exception: `apigateway.ErrCodeNotFoundException`